modules/gateways/xendit/lib/ActionBase.php

```diff
@@ -410,23 +410,16 @@ public function extractItems(\WHMCS\Billing\Invoice $invoice): array
      */
     public function extractCustomer(array $params, bool $isCreditCard = false): array
     {
-        $customerObject = [];
-
-        if (!empty($params['fullname'])) {
-            $customerObject['given_names'] = trim($params['fullname']);
-        } else {
-            if (!empty($params['firstname']) || !empty($params['lastname'])) {
-                $customerObject['given_names'] = trim(sprintf("%s %s", $params['firstname'], $params['lastname']));
-            }
-        }
-
-        if (!empty($params['email'])) {
-            $customerObject['email'] = $params['email'];
-        }
+        $customerObject = [
+            "given_names" => $params['firstname'],
+            "surname" => $params['lastname'],
+            "email" => $params['email'],
+            "mobile_number" => $params['phonenumber'],
+        ];
 
-        if (!empty($params['phonenumber'])) {
-            $customerObject['mobile_number'] = $params['phonenumber'];
-        }
+        $customerObject = array_filter($customerObject, function ($value){
```

Comment:
I think array_filter($customerObject) is enough.
Base on PHP doc, the default callback is `all empty entries of array will be removed`.

reference:
https://www.php.net/manual/en/function.array-filter.php

Reply:
yah. I made it shorter. @candrasaputra